### PR TITLE
Replace last uses of 'seconds' with ticks in world simulation

### PR DESCRIPTION
**Pull request by reaperrr**
This was long overdue.

### CODE REVIEW

**Comment by pchote** [comment]
I'm wondering if it would make more sense to keep the announcement intervals in seconds and change the code to use real seconds rather than ticks - see #18401.

**Comment by reaperrr** [comment]
Probably makes sense... though we have a few similar places that already use ticks, so we'd have to convert those to real seconds as well if we want to be consistent about it.

**Comment by reaperrr** [comment]
> Probably makes sense... though we have a few similar places that already use ticks, so we'd have to convert those to real seconds as well if we want to be consistent about it.

I will handle that in a separate PR, so removed the announcement/notifier changes from here.

Also added rename for `ScaredyCat.PanicLength` to `PanicDuration`, part for consistency and part to find/remember it more easily when we go for the lower default tick rate.

**Comment by pchote** [comment]
On that note, shouldn't we rename the crate `Lifetime` to `Duration` too?

**Comment by reaperrr** [comment]
> On that note, shouldn't we rename the crate Lifetime to Duration too?

Done. Haven't tested the updated update rule, though.

File: OpenRA.Mods.Common/UpdateRules/Rules/20201213/ReplaceCrateSecondsWithTicks.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class ReplaceCrateSecondsWithTicks : UpdateRule
+	{
+		public override string Name => "Changed Crate Lifetime to Duration and use ticks and renamed PanicLength to PanicDuration.";
+
+		public override string Description =>
+			"Crate.Lifetime was the last non-sound-related place to still use 'seconds'\n" +
+			" by multiplying with 25 internally. Converted to use ticks like everything else.\n" +
```
Comment (pchote) [inline OpenRA.Mods.Common/UpdateRules/Rules/20201213/ReplaceCrateSecondsWithTicks.cs]: Were these spaces at the start of the line intentional?

They look odd in the output:
```
Found 1 API change:
  * ReplaceCrateSecondsWithTicks: Changed Crate Lifetime to Duration and use ticks and renamed PanicLength to PanicDuration.
     Crate.Lifetime was the last non-sound-related place to still use 'seconds'
      by multiplying with 25 internally. Converted to use ticks like everything else.
      Also renamed Lifetime to Duration and ScaredyCat.PanicLength to PanicDuration to match other places.
```

**Review by pchote — COMMENTED** [review]
One small nit otherwise LGTM.

I have tested that the updated update rule behaves correctly.

File: OpenRA.Mods.Common/UpdateRules/Rules/20201213/ReplaceCrateSecondsWithTicks.cs (hunk 1 above)
Comment (reaperrr) [reply OpenRA.Mods.Common/UpdateRules/Rules/20201213/ReplaceCrateSecondsWithTicks.cs]: Kind of, I somehow (mis?)remembered a recent PR having something similar.
Doesn't really make sense with `\n` now that I think about it, I'll change it.

**Comment by reaperrr** [comment]
Updated.

**Review by pchote — DISMISSED** [review]

File: mods/ra/maps/fall-of-greece-1-personal-war/rules.yaml [hunk 2]
```diff
@@ -70,7 +70,7 @@ TRUK:
 
 HEALCRATE:
 	Crate:
-		Lifetime: 0
+		Duration: 0
```
Comment (Smittytron) [inline mods/ra/maps/fall-of-greece-1-personal-war/rules.yaml]: These mission overrides were consolidated into campaign-rules.yaml which will need to be changed to `Duration`.

**Comment by reaperrr** [comment]
Rebased.

**Review by teinarss — APPROVED** [review]

**Comment by teinarss** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/f27b819ddcc91bc65aa77637535eb8b4e619d26a...1e4f5783affbb5b8101763456c9f654a30cbdc29

**Pull request merged**